### PR TITLE
Add stale action

### PR DESCRIPTION
**Pull request by twsl**
As part of #1749 this PR add a stale action

### CODE REVIEW

File: .github/workflows/stale.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# This workflow warns and then closes issues and PRs that have had no activity for a specified amount of time.
+#
+# You can adjust the behavior by modifying this file.
+# For more information, see:
+# https://github.com/actions/stale
+name: "Close stale issues and PRs"
+
+on:
+  schedule:
+    - cron: "30 1 * * *"
```
Comment (makseq) [inline .github/workflows/stale.yml:10]: Can we change it for six months? (this is a little funny, but unfortunately, as it is)

File: .github/workflows/stale.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  pull-requests: write
+
+jobs:
+  stale:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/stale@v4
+        with:
+          stale-issue-message: "This issue is stale because it has been open 30 days with no activity. Remove stale label or comment or this will be closed in 5 days."
```
Comment (makseq) [inline .github/workflows/stale.yml:23]: Could we leave it only for PRs please?

**Review by makseq — CHANGES_REQUESTED** [review]

File: .github/workflows/stale.yml (hunk 1 above)
Comment (twsl) [reply .github/workflows/stale.yml:10]: This means it runs every day to check.
```
    #        ┌───────────── minute (0 - 59)
    #        │  ┌───────────── hour (0 - 23)
    #        │  │ ┌───────────── day of the month (1 - 31)
    #        │  │ │ ┌───────────── month (1 - 12 or JAN-DEC)
    #        │  │ │ │ ┌───────────── day of the week (0 - 6 or SUN-SAT)
    #        │  │ │ │ │
    #        │  │ │ │ │
    #        │  │ │ │ │
    #        *  * * * *
    - cron: "30 1 * * *"
```

File: .github/workflows/stale.yml (hunk 2 above)
Comment (twsl) [reply .github/workflows/stale.yml:23]: So do you want to increase the time for issues to 6 months or remove it?

File: .github/workflows/stale.yml (hunk 1 above)
Comment (makseq) [reply .github/workflows/stale.yml:10]: Yes, I just wanted to ask you to do it :)

File: .github/workflows/stale.yml (hunk 2 above)
Comment (makseq) [reply .github/workflows/stale.yml:23]: Remove for issues, they are all very valuable

**Review by makseq — DISMISSED** [review]

File: .github/workflows/stale.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+          close-issue-message: "This issue was closed because it has been stalled for 5 days with no activity."
+          days-before-issue-stale: 30
+          days-before-issue-close: -1 # never closed automatically
+          stale-issue-label: "no-issue-activity"
+          exempt-issue-labels: "awaiting-approval,work-in-progress"
+          stale-pr-message: "This PR is stale because it has been open 45 days with no activity. Remove stale label or comment or this will be closed in 10 days."
```
Comment (farioas) [inline .github/workflows/stale.yml:38]: 172 days

File: .github/workflows/stale.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+          close-pr-message: "This PR was closed because it has been stalled for 10 days with no activity."
```
Comment (farioas) [inline .github/workflows/stale.yml:39]: 182 days

File: .github/workflows/stale.yml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+          days-before-pr-stale: 45
```
Comment (farioas) [inline .github/workflows/stale.yml:40]: 172

File: .github/workflows/stale.yml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+          days-before-pr-close: 10
```
Comment (farioas) [inline .github/workflows/stale.yml:41]: 182

**Comment by farioas** [comment]
Added few comments regarding PR's:
from the date of creation - 172 days - if no activity mark as stale and grant 10 days for activity
from the date of creation - 182 days - if no activity for the last 10 days(until it's marked as stale and 182 days since it was opened - close it

**Review by farioas — CHANGES_REQUESTED** [review]
Please adjust according to my comments

File: .github/workflows/stale.yml (hunk 3 above)
Comment (twsl) [reply .github/workflows/stale.yml]: Done

File: .github/workflows/stale.yml (hunk 5 above)
Comment (twsl) [reply .github/workflows/stale.yml]: Done

**Review by farioas — APPROVED** [review]

**Pull request merged**